zenoh/src/api/key_expr.rs

```diff
@@ -272,6 +272,30 @@ impl<'a> KeyExpr<'a> {
             Ok(r.into())
         }
     }
+
+    pub(crate) fn keyexpr_intersect<'b, L, R>(left: L, right: R) -> bool
+    where
+        L: TryInto<KeyExpr<'a>>,
```

Comment:
In fact this is essentially done, to return false in case conversion to key expression fails, to avoid the boilerplate code which this function replaces.

Reply:
A bug in malformed input will be very hard to detect in this way without even a `debug` or `trace` log...

Reply:
This is exactly how it was done in the code that this function replaced, i.e as unwrap_or(false). In that context it was justified since data was coming from a verified source. This function is not public and the way it works makes it the most useful.  I caller really cares about sanitizing his input he can just manually call TryInto and then KeyExpr::intersects().  I think though it might be worth mentioning that conversion failures are treated as false.

Reply:
Explained that conversion failures are treated as false in function docs.

Reply:
Again, if malformed input is not expected, I'd simply add a log. 
If it was done like that before it does not mean it was 100% correct :)

Reply:
```sh
 git grep keyexpr_intersect
```
gives
```sh
zenoh/src/api/key_expr.rs:    pub(crate) fn keyexpr_intersect<'b, L, R>(left: L, right: R) -> bool
zenoh/src/net/routing/hat/client/pubsub.rs:                    .any(|sub| KeyExpr::keyexpr_intersect(sub.expr(), expr.full_expr()))
zenoh/src/net/routing/hat/client/pubsub.rs:                .any(|sub| KeyExpr::keyexpr_intersect(sub.expr(), key_expr))
zenoh/src/net/routing/hat/client/queries.rs:                    .any(|qbl| KeyExpr::keyexpr_intersect(qbl.expr(), expr.full_expr()))
zenoh/src/net/routing/hat/client/queries.rs:                    false => KeyExpr::keyexpr_intersect(qbl.expr(), key_expr),
zenoh/src/net/routing/hat/p2p_peer/pubsub.rs:                    .any(|sub| KeyExpr::keyexpr_intersect(sub.expr(), expr.full_expr()))
zenoh/src/net/routing/hat/p2p_peer/queries.rs:                    .any(|sub| KeyExpr::keyexpr_intersect(sub.expr(), expr.full_expr()))
```

This means the `keyexpr_intersect` is using in the routing logic to compute interesections.
I don't believe malformed input should be accepted at this stage. And if it is, I find it suspicious.
Without doing a full review of the whole routing logic, I believe that having a log will already help in identifying any potential issue. @OlivierHecart ?

Reply:
Let's have it as an error then.

Reply:
add error log to corresponding functions